code/datums/elements/area_based_godmode.dm

```diff
@@ -0,0 +1,82 @@
+/**
+ * Area-based godmode.
+ * (area, allow_area_subtypes, gain_message, lose_message)
+ */
+/datum/component/area_based_godmode
```

Comment:
This is a component in the ``elements`` folder

Reply:
~~Does it even hold state?~~

Reply:
I asked before because of that list. it may be useful to take advantage of dupe_mode tho